datatypes/src/error.rs

```diff
@@ -13,8 +13,31 @@ use crate::{
 
 pub trait ErrorSource: std::error::Error + Send + Sync + 'static + AsErrorSource {}
 
+impl ErrorSource for dyn std::error::Error + Send + Sync + 'static {}
+
 impl<T> ErrorSource for T where T: std::error::Error + Send + Sync + 'static {}
 
+pub trait BoxedResultExt<T, E>: Sized {
```

Comment:
what is this and why is it needed?

Reply:
this provides the context method for boxed sources